regimes/au/invoices.go

```diff
@@ -0,0 +1,97 @@
+package au
+
+import (
+	"github.com/invopop/gobl/bill"
+	"github.com/invopop/gobl/l10n"
+	"github.com/invopop/gobl/num"
+	"github.com/invopop/gobl/org"
+	"github.com/invopop/gobl/tax"
+	"github.com/invopop/validation"
+)
+
+type invoiceValidator struct {
+	inv *bill.Invoice
+}
+
+func validateInvoice(inv *bill.Invoice) error {
+	v := &invoiceValidator{inv: inv}
+	return v.validate()
+}
+
+func (v *invoiceValidator) validate() error {
+	inv := v.inv
+	return validation.ValidateStruct(inv,
+		validation.Field(&inv.Type,
+			validation.In(
+				bill.InvoiceTypeStandard,
+				bill.InvoiceTypeCreditNote,
+				bill.InvoiceTypeDebitNote,
+			),
+			validation.Skip,
+		),
+		validation.Field(&inv.IssueDate,
+			validation.Required,
+			validation.Skip,
+		),
+		validation.Field(&inv.Supplier,
+			validation.By(v.supplier),
+			validation.Skip,
+		),
+		// Customer ID necessary when total over AUD 1000
+		validation.Field(&inv.Customer,
+			validation.When(
+				inv.Totals.Total.Compare(num.MakeAmount(1000, 0)) == 1,
```

Comment:
This is a great idea, but a few caveats which mean its best avoided. We haven't added any of these types of amount based validations as they change over time. The only way to implement it correctly would be with some kind of time database. The other issue is that there are *always* exceptions, its very hard to generalise, so its best to avoid.

In terms of technical implementation, there are a few gotchas:
* the `num.MakeAmount(1000, 0)` ideally should be a constant somewhere.
* if the `Totals` or `Total` values are nil, this will cause a panic. Unlikely, but this is validation, so anything can happen!